### ISSUE TITLE
[request] Add played songs to the autoplaylist

### ISSUE THREAD

**Issue by rfsimoes** [Just-Some-Bots/MusicBot#615]
I'd like to be able to add the played songs to the autoplaylit.
Greate work btw.

**Comment by mdragon159** [comment]
This is actually a pretty darn great idea. If I'm free in the next week, I may personally work on this then put in a PR (but no promises there)

**Comment by imayhaveborkedit** [comment]
This is already on my todo list.  All of the new commands and command related changes will be the last thing I do for this update.

**Comment by mdragon159** [comment]
Sweet, thanks a ton! ^_^

**Comment by rfsimoes** [comment]
I've done it myself. It's probably not the best way to do it, but I did what I could.

**Comment by Injusticecraft** [comment]
I would really like to see this in a future update

**Comment by NNTin** [comment]
While at it also the ability for users to remove. Looking forward to this! :)
edit:
And option to only cache songs that are in the autoplaylist.

**Comment by eburch2** [comment]
self.autoplaylist.append(song_url)
            write_file(self.config.auto_playlist_file, self.autoplaylist)

This is a naive way of adding, I have it so everytime I queue up a song using !play it adds to the auto play list (this will weight the playlist, the more you request a song, the higher chance it will be randomly selected). Tried to figure out a way to unweight it, would recommend using entry's is_downloaded. That's where I looked but couldn't quite get it, so as of now I'm settling with a weighted list. Best of luck.

**Comment by Noxilex** [comment]
I've done it previously on my server. I added a `!list <url>` command which would work like the `!skip` command. What I mean by that is: Everytime someone wants to add a song to the playlist, a vote start, if the vote is won, then the song will be added to the playlist. (it also checks that the URL is not already in the playlist file)
I did it some time ago though (approx. 1 year), so it may need to be adapted to the current code. 
But I can share the files if anyone is interested.
